### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by mato2593**
This PR includes the chat implementation and handling new message notifications.

https://trello.com/c/bssH3S6u/13-5-as-a-user-i-should-be-able-to-start-a-conversation-with-any-of-my-compatible-targets

chat

### CODE REVIEW

**Review by ericnpc-zz — APPROVED** [review]

File: Target-MVD-Mathias/AppDelegate.swift [hunk 1]
```diff
@@ -79,18 +80,84 @@ class AppDelegate: UIResponder, UIApplicationDelegate, PushNotificationDelegate
...
   func onPushAccepted(_ pushManager: PushNotificationManager!, withNotification pushNotification: [AnyHashable : Any]!, onStart: Bool) {
     if let pushBody = pushNotification["u"] as? String {
       let json = JSON(parseJSON: pushBody)
       
       if json["title"].string != nil {
-        let match = Match.parse(fromJSON: json)
-        let chatViewController = UIStoryboard.instantiateViewController(ChatViewController.self)
-        window?.rootViewController?.present(chatViewController!, animated: false, completion: nil)
+        processNewMatchNotification(json)
+      } else if json["text"].string != nil {
```
Comment (mjurfest) [inline Target-MVD-Mathias/AppDelegate.swift]: If you use `.string` and the result is nil, the app will crash. Try using `.stringValue` and if the result does not exist the value will be nil by default

File: Target-MVD-Mathias/Extensions/UIColorExtension.swift [hunk 2]
```diff
@@ -19,5 +19,7 @@ extension UIColor {
   static let macaroniAndCheese = UIColor(r: 239, g: 198, b: 56)
   static let transparentWhite = UIColor(r: 255, g: 255, b: 255, alpha: 0.7)
   static let brightSkyBlue = UIColor(r: 47, g: 188, b: 247)
-  
+  static let white70 = UIColor(r: 244, g: 244, b: 244, alpha: 0.7)
```
Comment (mjurfest) [inline Target-MVD-Mathias/Extensions/UIColorExtension.swift]: Maybe you can add some "customization" to these colors. For example: 
`tWhite70`, so you know that the color is from Target. WDYT?

**Review by mjurfest — COMMENTED** [review]
Small comment

File: Target-MVD-Mathias/AppDelegate.swift (hunk 1 above)
Comment (mato2593) [inline Target-MVD-Mathias/AppDelegate.swift:101]: It's the other way around, `.string` is the optional value.

File: Target-MVD-Mathias/Extensions/UIColorExtension.swift (hunk 2 above)
Comment (mato2593) [reply Target-MVD-Mathias/Extensions/UIColorExtension.swift]: I think that's a good idea, I'll do it.

**Review by pMalvasio — APPROVED** [review]

**Review by mjurfest — APPROVED** [review]

**Pull request merged**